python/hail/expr/functions.py

```diff
@@ -1778,7 +1778,159 @@ def rand_unif(min, max, seed=None) -> Float64Expression:
     -------
     :class:`.Expression` of type :py:data:`.tfloat64`
     """
-    return _seeded_func("runif_seeded", tfloat64, seed, min, max)
+    return _seeded_func("rand_unif", tfloat64, seed, min, max)
+
+
+@typecheck(a=expr_float64,
+           b=expr_float64,
+           min=nullable(expr_float64),
+           max=nullable(expr_float64),
+           seed=nullable(int))
+def rand_beta(a, b, min=None, max=None, seed=None) -> Float64Expression:
+    """Returns a random floating-point number drawn from a beta distribution
+    with parameters a and b.
+
+    Notes
+    -----
+    The optional parameters `min` and `max` represent a truncated beta
+    distribution with parameters a and b and support `[min, max]`. Draws are
+    made via rejection sampling, i.e. returning the first draw from Beta(a,b)
+    that falls in range `[min, max]`. This procedure may be slow if the
+    probability mass of Beta(a,b) over `[min, max]` is small.
+
+    Examples
+    --------
+
+    >>> hl.rand_beta(0, 1).value
+    0.6696807666871818
+
+    >>> hl.rand_beta(0, 1).value
+    0.8512985039011525
+
+    Parameters
+    ----------
+    a : float or :class:`.Expression` of type :py:data:`.tfloat64`
+    b : float or :class:`.Expression` of type :py:data:`.tfloat64`
+    min : float or :class:`.Expression` of type :py:data:`.tfloat64`
+        (Optional) Left boundary of truncated beta distribution.
+    max : float or :class:`.Expression` of type :py:data:`.tfloat64`
+        (Optional) Right boundary of truncated beta distribution.
+    seed : :obj:`int` or `None`
+        If not `None`, function will be seeded with provided seed.
+
+    Returns
+    -------
+    :class:`.Expression` of type :py:data:`.tfloat64`
+    """
+    if min is None and max is None:
+        return _seeded_func("rand_beta", tfloat64, seed, a, b)
+    if min is None:
+        min = hl.literal(0)
+    if max is None:
+        max = hl.literal(1)
+
+    return _seeded_func("rand_beta", tfloat64, seed, a, b, min, max)
+
+
+@typecheck(a=expr_float64,
+           scale=expr_float64,
+           seed=nullable(int))
+def rand_gamma(a, scale, seed=None) -> Float64Expression:
```

Comment:
I think it will be clearer to rename `a` to `shape`.

https://en.wikipedia.org/wiki/Gamma_distribution

Reply:
And make sure scale is really scale and not rate